benchmarking/smartseq2/compare_matrix/Compare_data_matrix.R

```diff
@@ -0,0 +1,380 @@
+opt <- parse_args(opt_parser)
+matrixfile1 <- opt$matrix1
+matrixfile2 <- opt$matrix2
+gtf_file <-  opt$gtf_file
+output_name <- opt$output_prefix
+# color palette
+palette(c("#00AFBB", "#E7B800"))
+# load data matrix
+mat1 <- read.csv(matrixfile1)
+mat2 <- read.csv(matrixfile2)
+# calculate folder changes
+fc <- foldchanges(mat1, mat2)
+# calucate mean or median FC cross samples
+fc.dd <- fc[,-c(1:2)]
+fc.m <- apply(fc.dd, 1, median)
+# boxplot
+# Box plot (bp)
+bxp <- ggviolin(
+  fc.m,
+  ylab = 'log FoldChanges',
+  xlab = '',
+  ggtheme = theme_minimal(),
+  add = "jitter"
+)
+bxp <- addTheme(bxp)
+# correlation
+# return two matrix of corraltion matrix
+# take diag of correlation matrix
+# plot into histogram
+mcor <- CorrDataMatrix(mat1, mat2, 0)
+phist <- gghistogram(
+  mcor,
+  x = "cor",
+  fill = "lightgray",
+  add = "mean",
+  rug = TRUE,
+  xlab = "Correlation"
+)
+phist <- addTheme(phist)
+# parse GTF
+genes <- ParseGene(gtf_file)
+# summary foldchanges by biotypes
+names(fc.m) <- fc[, 1]
+fc_tb <- summaryFoldChanges(fc.m, genes, 0.5)
+# SNN cluster
+mat1.snn <- RunSNNCluster(mat1)
+mat2.snn <- RunSNNCluster(mat2)
+# Calculate adjusted rand index between two cluster results
+# to figure out how similar the results are
+# ideal, if we habe pre-labeled data, we can calculate RAND index
+# between clustering result and labels
+rand <- adjustedRandIndex(mat1.snn$membership,mat2.snn$membership)
+# visualization
+mtb <- melt(fc_tb)
+pb <- ggbarplot(
+  mtb,
+  x = 'Var2',
+  y = 'value',
+  color = "Var1",
+  fill = "Var1",
+  # change fill color by FC
+  x.text.angle = 90,
+  # Rotate vertically x axis texts
+  xlab = "",
+  ylab = "",
+  rotate = TRUE,
+  ggtheme = theme_minimal(),
+  legend.title = "Summary of FC genes"
+)
+pb <- addTheme(pb)
+# combine plots
+text <- paste(
+  "A: Violinplot of log-foldchanges.",
+  "B: Histogram of correlation between data matrix",
+  "C: Barplot to summary log-foldchanges by gene biotyps",
+  sep = ' '
+)
+# visualize SNN clustering
+snn.p1 <- ggscatter(
+  mat1.snn,
+  x = "PC1",
+  y = "PC2",
+  color = "membership",
+  palette = "jco",
+  size = 8,
+  alpha = 0.6,
+  title = 'Base Pipeline'
+) + border()
+snn.p2 <- ggscatter(
+  mat2.snn,
+  x = "PC1",
+  y = "PC2",
+  color = "membership",
+  palette = "jco",
+  size = 8,
+  alpha = 0.6,
+  title = 'Updated Pipeline'
+) + border()
+snn.p1 <- addTheme(snn.p1)
+snn.p2 <- addTheme(snn.p2)
+
+p1 <- ggarrange(bxp,
+                phist,
+                labels = c("A", "B"),
+                ncol = 1,
+                nrow = 2)
+p2 <- ggarrange(pb, labels = c("C"))
+p3 <- ggarrange(p1,
+                p2,
+                ncol = 2,
+                nrow = 1)
+p4 <- ggparagraph(text,
+                  face = "bold",
+                  size = 20,
+                  color = "black")
+p5 <- ggarrange(p3,
+                p4,
+                ncol = 1,
+                nrow = 2,
+                heights = c(1, 0.1))
+text <- paste(
```

Comment:
I didn't see this text in the pdfs you linked in the ticket. Are they up to date?

Reply:
will check

Reply:
Didn't find them either :/